### ISSUE TITLE
Use mxj for GraphQL Client

### ISSUE THREAD

**Issue by sshaw** [bold-commerce/go-shopify#315]
Now that Shopify is eliminating its products and variants endpoints more folks will be using the GraphQL client.  Dealing with JSON in Go is not fun. Especially when the JSON object can vary like it does in GraphQL. Unless types are generated/imported from the GraphQL schema (bulky and maintenance overhead), users are going to have to write the same boilerplate `struct` definitions or the same `interface`/`any` type cast code (🙀) over and over. With Shopify using Relay's conventions of `edges` and `nodes` for relations, this makes these boilerplate structures a headache.

Instead of forcing callers to define these, I'm going to suggest using [mjx](https://github.com/clbanning/mxj). This makes handling the GraphQL response a breeze. The client returns an `mxj.Map` and the caller can do things like 

```go
nodes, _ := data.ValuesForPath("metafieldStorefrontVisibilities.edges.node")
```

Thoughts?

**Comment by oliver006** [comment]
This library won't transparently handle the GraphQL endpoints for you, you're better off using a proper GraphQL client if you need robust support for the /product and /variants endpoint

**Comment by sshaw** [comment]
> This library won't transparently handle the GraphQL endpoints for you, you're better off using a proper GraphQL client if you need robust support for the /product and /variants endpoint


Yes the endpoints are easy it's the response parsing that is a pain and that's where this library comes in.
